addOns/custompayloads/src/main/java/org/zaproxy/zap/extension/custompayloads/PayloadIterator.java

```diff
@@ -57,6 +58,9 @@ public boolean hasNext() {
 
         @Override
         public String next() {
+            if (!hasNext()) {
```

Comment:
This advances the internal iterator, which would cause the `PayloadIterator` to skip payloads. Could you add tests?